src/App.tsx

```diff
@@ -93,60 +109,79 @@ function App() {
     setCorrectAnswer(parseInt(answer));
   };
 
-  const handleOnInputFocus = (index: number) => {
-    generateEquation();
-    if (answers[index] === undefined) {
-      setInputEnabled((prev) => {
-        const updated = [...prev];
-        updated[index] = true;
-        return updated;
-      });
-      setModalOpen(true);
-    }
-  };
+    const handleOnInputFocus = (index: number) => {
+      setCurrentIndex(index);
+      generateEquation();
+      if (answers[index] === undefined) {
+        setInputEnabled((prev) => {
+          const updated = [...prev];
+          updated[index] = true;
+          return updated;
+        });
+        setModalOpen(true);
+      }
+    };
+
+    const startTimer = (index: number) => {
+      clearTimeout(timeoutRef.current);
+      timeoutRef.current = setTimeout(() => {
+        setFormInputs((prev) => ({
+          ...prev,
+          [Object.keys(formInputs)[index] as keyof Form]: "",
+        }));
+
+        const updatedAnswers = [...answers];
+        // @ts-ignore
+        updatedAnswers[index] = undefined;
+        setAnswers(updatedAnswers);
+      }, 10000);
+    };
```

Comment:
The first form field keeps refreshing even after filled within the supposed allowed 20 seconds.
You set 10 seconds instead of 20 seconds.

Reply:
I have changed the timer to 20 seconds but the issue you experienced was from an earlier update which has been resolved in the most recent updates. Please confirm that the behaviour is as intended now.